### ISSUE TITLE
Insecure clear_inactive_users endpoint

### ISSUE THREAD

**Issue by jankytech** [pluja/Parasitter#85]
In the settings panel there is a button to clear the inactive users if you are an admin. This just sends a get request to an endpoint that seems to take a hash of the admin password in order to authenticate the request. This is not a good way of keeping endpoints secure and essentially removes any security that hashing your passwords gives you. For example if the database got leaked and the hash of the admin user was visible then they could call this (and any other endpoint secured in the same way) without needing to work out what your admin password is and just by sending the hash to the endpoint:

From settings.html
```
<a href="/clear_inactive_users/{{current_user.password_hash}}"><button class="ui red button">Delete</button></a></div>
```

**Comment by jankytech** [comment]
A better solution would be to verify which user is logged in via the session and if you wanted to make sure then ask for a password that gets sent in a post request then hashed on the server side and matched against the database stored value for the user's password  hash.

**Comment by FireMasterK** [comment]
You're right, currently, you just need to be logged in and have the hash of the password to delete all users but the problem still persists. Ideally, we should use a form with CSRF protection and require the password instead of the hash so it becomes much harder to perform malicious activities in case of breaches.

**Comment by pluja** [comment]
I have this in mind, I really knew this was insecure from point 0 of developing it. I wanted a quick solution and this seemed like a good one. I will disable this option for now, and I will develop it better in a future version. Both of you are right.

IMO the best option for this should be prompting the password when executing this option, hashing it and matching as @mansguiche  said. I think that flask WTForms already have CSRF protection by default @FireMasterK and I'm using WTForms on most important forms of the app.

Right now I have a very limited free time, and I will only be solving bugs and errors. If anyone wants to contribute is welcome, but I can't say when I will be able to dedicate enough time to develop the web-app.
